### PR TITLE
Feature: Display recent searches when clicking on the search box

### LINKED ISSUE #8104: add search history to search box
### What's the Problem?

add search history to the search box and also a button to remove searches

### Solution/Idea

add search history to the search box and also a button to remove searches

### Alternatives

add search history to the search box and also a button to remove searches

### Priorities

| Capability | Priority |
| :---------- | :------- |
| This proposal will allow developers to accomplish W | Must |
| This proposal will allow end users to accomplish X | Should |
| This proposal will allow developers to accomplish Y | Could |
| This proposal will allow end users to accomplish Z | Won't |

### Files Version

2.1.5

### Windows Version

w10 21h1 64 bits

### PR DESCRIPTION
**Pull request by ferrariofilippo**
**Resolved / Related Issues**
- Related #9793 (Part 1 and 3)
- closes #8104

**Details of Changes**
- The SearchBox is no more cleaned after a query is submitted
- When the SearchBox is focused but empty, it shows the list of latest queries submitted. I limited this number to 3 to guarantee better performances. (Should I remove the file icon from these entries?)

**Comments**
- I couldn't implement the second feature requested, as this depends on a Windows Control

**Validation**
- Built and ran the app

**Screenshots**
Screenshot (32)
Screenshot (31)

### CODE REVIEW

File: src/Files.App/UserControls/SearchBox.xaml
Comment (yaichenbaum) [inline src/Files.App/UserControls/SearchBox.xaml:77]: Can we bind this event directly to the view model?

File: src/Files.App/ViewModels/SearchBoxViewModel.cs
Comment (yaichenbaum) [inline src/Files.App/ViewModels/SearchBoxViewModel.cs:96]:
```suggestion
                if (oldQueries.Count > 5)
```

**Comment by verdaderoken** [comment]
image
For me, I think it might be better if the icon for the old queries is something like these search history icons:
image

**Comment by yaichenbaum** [comment]
>  (Should I remove the file icon from these entries?)

@ferrariofilippo what are your thoughts on using the icon in the screenshot from @verdaderoken?

**Comment by ferrariofilippo** [comment]
Ok, I'll fix that

**Comment by yaichenbaum** [comment]
Should just be a regular glyph `&#xE81C;`

**Review by yaichenbaum — DISMISSED** [review]
Excellent improvement!

I'd like to wait for another review before merging.

**Comment by ferrariofilippo** [comment]
Ok np

File: src/Files.App/UserControls/SearchBox.xaml
Comment (yaichenbaum) [inline src/Files.App/UserControls/SearchBox.xaml:61]: Is it possible to use a converter here? It would be nice to set the glyphs directly from the xaml. Alternatively, a comment with the two possible glyphs would also work.

File: src/Files.App/Converters/BoolToGlyph.cs
Comment (yaichenbaum) [inline src/Files.App/Converters/BoolToGlyph.cs:7]: I should have specified, the values for the converter should be in the xaml file as a resource. It might be easier to just add a comment to the xaml file and skip using a converter all together.

File: src/Files.App/Converters/BoolToGlyph.cs
Comment (d2dyno1) [inline src/Files.App/Converters/BoolToGlyph.cs:3]: Please rename the file as well

**Review by d2dyno1 — CHANGES_REQUESTED** [review]

File: src/Files.App/Converters/BoolToGlyph.cs
Comment (d2dyno1) [reply src/Files.App/Converters/BoolToGlyph.cs:3]: Honestly, I'd make it a generic glyph converter where the converter parameter is the glyph:
`<glyphIfTrue>|<glyphIfFalse>`

Comment (ferrariofilippo) [reply src/Files.App/Converters/BoolToGlyph.cs:3]: Ok, so I'll change this and I'll add a comment in the xaml to explian the two glyphs

Comment (yaichenbaum) [reply src/Files.App/Converters/BoolToGlyph.cs:3]: > Honestly, I'd make it a generic glyph converter where the converter parameter is the glyph:

That's what I originally intended, thought we had a converter for this already.

Comment (ferrariofilippo) [reply src/Files.App/Converters/BoolToGlyph.cs:3]: So, should I use the built-in converter (like in the picture) or should I add a brand new converter?

Comment (yaichenbaum) [reply src/Files.App/Converters/BoolToGlyph.cs:3]: The built in one looks good, that's what I had in mind.

File: src/Files.App/Filesystem/ListedItem.cs
Comment (yaichenbaum) [inline src/Files.App/Filesystem/ListedItem.cs:68]: Does this belong in ListedItem?

Comment (ferrariofilippo) [reply src/Files.App/Filesystem/ListedItem.cs:68]: Yes, I added it to differentiate recent searches from other `ListedItem`s (in order to show the correct glyph).

Comment (yaichenbaum) [reply src/Files.App/Filesystem/ListedItem.cs:68]: This is just me wondering but I don't think ListedItem is supposed to be coupled to search or have any knowledge of search.

Comment (ferrariofilippo) [reply src/Files.App/Filesystem/ListedItem.cs:68]: Do you have anything in mind to solve this problem?
To my mind, we could add a new class (called something like `ListedSuggestion`) that inherits from `ListedItem` and expands it with the `IsSearch` property.

Comment (yaichenbaum) [reply src/Files.App/Filesystem/ListedItem.cs:68]: Adding @d2dyno1

Comment (d2dyno1) [reply src/Files.App/Filesystem/ListedItem.cs:68]: Why do we even need to use/inherit from `ListedItem`? Isn't a search item a name and an icon?

Comment (ferrariofilippo) [reply src/Files.App/Filesystem/ListedItem.cs]: @d2dyno1

**Review by d2dyno1 — APPROVED** [review]

**Review by yaichenbaum — APPROVED** [review]

**Pull request merged**
